packages/transactions/src/__tests__/lifetime-test.ts

```diff
@@ -0,0 +1,122 @@
+import { Address } from '@solana/addresses';
+import { SOLANA_ERROR__TRANSACTION__NONCE_ACCOUNT_CANNOT_BE_IN_LOOKUP_TABLE, SolanaError } from '@solana/errors';
+import { Blockhash } from '@solana/rpc-types';
+import {
+    CompiledTransactionMessage,
+    CompiledTransactionMessageWithLifetime,
+    Nonce,
+} from '@solana/transaction-messages';
+
+import { getTransactionLifetimeConstraintFromCompiledTransactionMessage } from '../lifetime';
+
+const SYSTEM_PROGRAM_ADDRESS = '11111111111111111111111111111111' as Address;
+const U64_MAX = 2n ** 64n - 1n;
+
+describe('getTransactionLifetimeConstraintFromCompiledTransactionMessage', () => {
+    it('returns a blockhash transaction lifetime when there are no instructions', () => {
+        const compiledTransactionMessage = {
+            instructions: [],
+            lifetimeToken: 'abc',
+        } as unknown as CompiledTransactionMessage & CompiledTransactionMessageWithLifetime;
+
+        const lifetime = getTransactionLifetimeConstraintFromCompiledTransactionMessage(compiledTransactionMessage);
+        expect(lifetime).toEqual({ blockhash: 'abc' as Blockhash, lastValidBlockHeight: U64_MAX });
+    });
+
+    describe('returns a blockhash transaction lifetime when the first instruction is not an AdvanceNonceAccount instruction', () => {
```

Comment:
Maybe add tests that are like ‘because it has no `accountIndices` property’ and ‘because it has no `data` property.’